apps/src/blockly/addons/cdoBlockFlyout.ts

```diff
@@ -232,7 +239,7 @@ export default class CdoBlockFlyout extends GoogleBlockly.HorizontalFlyout {
         cursorX += blockHW.width + gaps[i];
 
         this.addBlockListeners_(root, block, rect);
-      } else if (item.type === 'button') {
+      } else if (item.type === 'button' && item.button) {
```

Comment:
I think that we can theoretically expect both of these conditions to be the same in terms of truthiness, but that it isn't made explicit by the `FlyoutItem` interface. Does that track?

https://github.com/google/blockly/blob/a517317b77c1e58727b748a99f9ae70b2c27d26a/core/flyout_base.ts#L1311-L1315

Reply:
Correct, sometimes TypeScript makes you put in explicit null checks even when you know the thing won't be null because of some other condition. Seems like I could replace this check with just `else if(item.button)`?

Reply:
Oh maybe! Either way, should we use the same convention here and at line 213 above?

```
      if (item.type === 'block') {
        const block = item.block;
        if (!block) {
          continue;
        }
...
```

Reply:
Oh wait, there's also this:

```
   */
  // TODO: once we upgrade to v10 of blockly we should not need this anymore
  // and can rely on the parent version of this function.
  // This is copied from the core blockly repo to include a fix from this PR on Blockly:
  // https://github.com/google/blockly/pull/7333
```

Reply:
Unfortunately that PR got reverted due to rtl issues (I updated the comment to this effect). As far as I can tell the fix from the pr is still fine for this use case, even in rtl, probably because this is a tiny flyout.
And yes, good catch on line 213, will update both.

Reply:
Should we change the link from the reverted PR to the open issue so we can track more easily?

Reply:
https://github.com/google/blockly/issues/6280

Reply:
updated comment and fixed logic!

Reply:
LGTM!